src/main/java/edu/harvard/iq/dataverse/DataFile.java

```diff
@@ -398,6 +399,29 @@ public Long getOriginalFileSize() {
         }
         return null;
     }
+    
+    public String getOriginalFileName() {
+        if (isTabularData()) {
+            DataTable dataTable = getDataTable();
+            if (dataTable != null) {
+                return dataTable.getOriginalFileName() != null ? dataTable.getOriginalFileName()
+                        : getDerivedOriginalFileName();
+            }
+        }
+        return null;
+    }
+
+    private String getDerivedOriginalFileName() {
+        FileMetadata fm = getFileMetadata();
+        String filename = fm.getLabel();
+        String originalExtension = FileUtil.generateOriginalExtension(getOriginalFileFormat());
+        String extensionToRemove = StringUtil.substringIncludingLast(filename, ".");
+        if (StringUtil.nonEmpty(extensionToRemove)) {
+            String newFileName = filename.replace(extensionToRemove, originalExtension);
+            return newFileName;
+        }
```

Comment:
@sekmiller Am I reading this correctly - that if the current filename has no extension ("foo"), this code will return "foo"? - shouldn't it still return "foo.<ORIG_EXT>"? 
(this shouldn't really happen - since ingested files should always have the ".tab" extension - but still)

Reply:
Good points. I will take another pass at this.

Reply:
@landreev I put in your suggestion on using replaceAll and relying on the fact that a tabular file should always end in ".tab ". It occurs to me that any editing of the file label to anything other than foo.tab will break this (including your "no extension" case). How worried should we be about this, and any suggestions to code against it?

Reply:
Answered my own question while out for a run: so the replaceAll("tab$" which I put in should fix 100% of the cases. Since we know it won't I can modify the previous "look for an extension to replace" code by adding a $ to the replace string so that inadvertent internal instances of the extension are not replaced, and if there's no extension found just append the extension derived from the file type. All of this points out that even after all this we cannot be certain that we're returning anything like the actual original file name.